internal/services/appservice/helpers/function_app_schema.go

```diff
@@ -0,0 +1,904 @@
+package helpers
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/web/mgmt/2021-02-01/web"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	apimValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+const (
+	StorageStringFmt = "DefaultEndpointsProtocol=https;AccountName=%s;AccountKey=%s;EndpointSuffix=%s"
+)
+
+type SiteConfigLinuxFunctionApp struct {
+	AlwaysOn                      bool                               `tfschema:"always_on"`
+	AppCommandLine                string                             `tfschema:"app_command_line"`
+	ApiDefinition                 string                             `tfschema:"api_definition_url"`
+	ApiManagementConfigId         string                             `tfschema:"api_management_api_id"`
+	AppInsightsInstrumentationKey string                             `tfschema:"application_insights_key"` // App Insights Instrumentation Key
+	AppInsightsConnectionString   string                             `tfschema:"application_insights_connection_string"`
+	AppScaleLimit                 int                                `tfschema:"app_scale_limit"`
+	UseManagedIdentityACR         bool                               `tfschema:"container_registry_use_managed_identity"`
+	ContainerRegistryMSI          string                             `tfschema:"container_registry_managed_identity_client_id"`
+	DefaultDocuments              []string                           `tfschema:"default_documents"`
+	ElasticInstanceMinimum        int                                `tfschema:"elastic_instance_minimum"`
+	Http2Enabled                  bool                               `tfschema:"http2_enabled"`
+	IpRestriction                 []IpRestriction                    `tfschema:"ip_restriction"`
+	LoadBalancing                 string                             `tfschema:"load_balancing_mode"` // TODO - Valid for FunctionApps?
```

Comment:
The service honours the value, and typically blocks the use of settings not supported by Functions, but no, I've not conclusively proven that this has meaningful impact on connectivity to hosted Functions. I have no strong opinion on keeping / removing and deprecating / re-introducing respectively as necessary? (afaik, this can only be relevant in Standard and Premium plans, and would be ignored in Consumption and ElasticPremium)